packages/react-native-reanimated/apple/LayoutReanimation/REASharedTransitionManager.m

```diff
@@ -351,26 +361,69 @@ - (NSArray *)sortViewsByTags:(NSArray *)views
+
+  if (!(maybeTabNavigatorForSharedView && maybeTabNavigatorForSiblingView) ||
+      maybeTabNavigatorForSharedView != maybeTabNavigatorForSiblingView) {
+    return siblingView;
+  }
+
+  NSArray<NSNumber *> *sharedGroup = _getSharedGroupBlock(sharedView.reactTag);
+  int siblingIndex = [sharedGroup indexOfObject:siblingView.reactTag];
+  REAUIView *activeTab = [REAScreensHelper getActiveTabForTabNavigator:maybeTabNavigatorForSharedView];
+  for (int i = siblingIndex; i >= 0; i--) {
+    NSNumber *viewTag = sharedGroup[i];
+    REAUIView *view = [_animationManager viewForTag:viewTag];
+    if ([REAScreensHelper isView:view ChildOfScreen:activeTab]) {
+      return view;
+    }
+  }
+  return nil;
+}
+
+- (REAUIView *)getTabNavigator:(REAUIView *)view
+{
+  REAUIView *currentView = view;
+  while (currentView.superview) {
+    if ([currentView isKindOfClass:NSClassFromString(@"RNSScreenNavigationContainerView")]) {
+      return currentView;
+    }
+    currentView = (REAUIView *)currentView.superview;
+  }
+
+  currentView = view;
+  while (currentView.reactSuperview) {
+    if ([currentView isKindOfClass:NSClassFromString(@"RNSScreenNavigationContainerView")]) {
+      return currentView;
+    }
+    currentView = (REAUIView *)currentView.reactSuperview;
+  }
+  return nil;
+}
+
 /*
   Method swizzling is used to get notification from react-native-screens
   about push or pop screen from stack.
 */
 - (void)swizzleScreensMethods
 {
 #if LOAD_SCREENS_HEADERS
-  static dispatch_once_t onceToken;
-  dispatch_once(&onceToken, ^{
-    SEL viewDidLayoutSubviewsSelector = @selector(viewDidLayoutSubviews);
-    SEL notifyWillDisappearSelector = @selector(notifyWillDisappear);
-    SEL viewIsAppearingSelector = @selector(viewIsAppearing:);
-    Class screenClass = [RNSScreen class];
-    Class screenViewClass = [RNSScreenView class];
-    BOOL allSelectorsAreAvailable = [RNSScreen instancesRespondToSelector:viewDidLayoutSubviewsSelector] &&
-        [RNSScreenView instancesRespondToSelector:notifyWillDisappearSelector] &&
-        [RNSScreen instancesRespondToSelector:viewIsAppearingSelector] &&
-        [RNSScreenView instancesRespondToSelector:@selector(isModal)]; // used by REAScreenHelper
-
-    if (allSelectorsAreAvailable) {
+  SEL viewDidLayoutSubviewsSelector = @selector(viewDidLayoutSubviews);
+  SEL notifyWillDisappearSelector = @selector(notifyWillDisappear);
+  SEL viewIsAppearingSelector = @selector(viewIsAppearing:);
+  Class screenClass = [RNSScreen class];
+  Class screenViewClass = [RNSScreenView class];
+  BOOL allSelectorsAreAvailable = [RNSScreen instancesRespondToSelector:viewDidLayoutSubviewsSelector] &&
+      [RNSScreenView instancesRespondToSelector:notifyWillDisappearSelector] &&
+      [RNSScreen instancesRespondToSelector:viewIsAppearingSelector] &&
+      [RNSScreenView instancesRespondToSelector:@selector(isModal)]; // used by REAScreenHelper
+
+  if (allSelectorsAreAvailable) {
```

Comment:
Why did you move it like this?

Reply:
I resolved some issues related to missing animations after reload by ensuring that the instance of `_isConfigured` is set to true every time after the `swizzleScreensMethods` call, upon completing the swizzling process (it happens only once for the class).